### PR TITLE
Mute, Unmute, Leave Audio

### PR DESCRIPTION
**Pull request by mmarangoni**
- Join Audio button switches to Leave audio after joining in audio or listen only
- Joining in audio will display the mute/unmute button options (listen only has no muting options)
- Mute button switches to unmute depending on status
- Leave audio properly exits the call, user is able to rejoin after exiting

### CODE REVIEW

File: bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/container.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+class MuteAudioContainer extends React.Component {
+  muteUser() {
+    callServer('muteUser', Auth.userID);
+  }
+
+  unmuteUser() {
+    callServer('unmuteUser', Auth.userID);
+  }
+
+  render() {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/container.jsx]: DRY
```
render() {
  const { isMuted } = this.props;
  const onClick = this.muteUser;
  const label = 'Mute';

  if(isMuted) {
    onClick = this.unmuteUser;
    label = 'Unmute';
  }
  
  return (
    <Button
      onClick={onClick}
      label={label}
      color={'primary'}
      icon={'audio'}
      size={'lg'}
      circle={true}
    />
  )
}
```

File: bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/container.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react';
+import {createContainer} from 'meteor/react-meteor-data';
+import {callServer} from '/imports/ui/services/api';
+import Button from '/imports/ui/components/button/component';
+import Users from '/imports/api/users/index';
+import Auth from '/imports/ui/services/auth/index';
+
+class MuteAudioContainer extends React.Component {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/container.jsx]: > A container does data fetching and then renders its corresponding sub-component. That’s it.

https://medium.com/@learnreact/container-components-c0e67432e005#.w69vhzi1t

File: bigbluebutton-html5/imports/ui/components/actions-bar/component.jsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ export default class ActionsBar extends Component {
         <div className={styles.center}>
           <JoinAudioContainer
             open={openJoinAudio.bind(this)}
-            close={exitAudio}
+            close={function () {exitAudio(function() {})}}
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/component.jsx]: The function `exitAudio` already has a noop as a default parameter.

**Comment by mmarangoni** [comment]
Linted and fixed multiple issues based on Oswaldo's feeback, please review.

File: bigbluebutton-html5/imports/ui/components/actions-bar/audio-menu/component.jsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { createContainer } from 'meteor/react-meteor-data';
+import Button from '/imports/ui/components/button/component';
+import Users from '/imports/api/users/index';
+import Auth from '/imports/ui/services/auth/index';
+import MuteAudioContainer from '../mute-button/container';
+
+export default class JoinAudioComponent extends React.Component {
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/audio-menu/component.jsx]: Don't need the `Component` suffix

File: bigbluebutton-html5/imports/ui/components/actions-bar/audio-menu/container.jsx [hunk 5]
```diff
@@ -1,57 +1,25 @@
...
+  render() {
+    return (
+      <JoinAudioComponent {...this.props} />
+    );
+  }
+}
+
+export default createContainer((params) => {
+  const data = {
+    isInAudio: Users.findOne({ userId: Auth.userID }).user.voiceUser.joined,
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/audio-menu/container.jsx]: Avoid fetching the user two times by assign it to a variable before.

```
const user = Users.findOne({ userId: Auth.userID }).user;

return {
    isInAudio: user.voiceUser.joined,
    isInListenOnly: user.listenOnly,
    open: params.open,
    close: params.close,
};
```

File: bigbluebutton-html5/imports/ui/components/actions-bar/component.jsx [hunk 6]
```diff
@@ -10,7 +10,7 @@ import Users from '/imports/api/users/index';
 import JoinAudioContainer from './audio-menu/container';
 import { exitAudio } from '/imports/api/phone';
 
-const openJoinAudio = () => {showModal(<Audio />)};
+const openJoinAudio = () => {showModal(<Audio />);};
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/component.jsx]: `const openJoinAudio = () => showModal(<Audio />);`

File: bigbluebutton-html5/imports/ui/components/actions-bar/component.jsx [hunk 7]
```diff
@@ -29,7 +29,11 @@ export default class ActionsBar extends Component {
         <div className={styles.center}>
           <JoinAudioContainer
             open={openJoinAudio.bind(this)}
-            close={exitAudio}
+            close={
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/component.jsx]: Everything in a single line would be easier to read

File: bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/component.jsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import {createContainer} from 'meteor/react-meteor-data';
+import {callServer} from '/imports/ui/services/api';
+import Button from '/imports/ui/components/button/component';
+import Users from '/imports/api/users/index';
+import Auth from '/imports/ui/services/auth/index';
+
+export default class MuteAudioComponent extends React.Component {
+  muteUser() {
+    callServer('muteUser', Auth.userID);
```
Comment (oswaldoacauan) [inline bigbluebutton-html5/imports/ui/components/actions-bar/mute-button/component.jsx]: You need to pass those callbacks as props from the container/service. The component shouldn't use `callSever` functions. You can take look into the Chat service, container and component.

https://github.com/bigbluebutton/bigbluebutton/blob/e26e9fd06029ea92c7b19769e4e532c2bf894eda/bigbluebutton-html5/imports/ui/components/chat/container.jsx#L82-L92

**Pull request merged**